### ISSUE TITLE
[Feature request] backlinks in What links here pane

### ISSUE THREAD

**Issue by omega3** [zadam/trilium#1299]
It would be great if there were backlinks in What links here pane.
In Demo note I created a link to sample Child note:
https://i.imgur.com/EgiaqRz.png

And when I am in Child note What links here pane shows link to Demo note. But when I am in Demo it doesn't show link to Child note:
https://i.imgur.com/HrmWXKk.png

~~Here is example how Obsidian handle internal links:
[Obsidian - The Power of Backlinks and the Knowledge Graph - Effective Remote Work](https://www.youtube.com/watch?v=n7xrHPpTWJ0)~~

I am not saying that it needs to be the same, but implementing the idea of **switching back and forth** could increase overall easiness of use of application. Using graph when there are too many links is not that attractive to me.

**Comment by zadam** [comment]
I'm a bit confused - "What links here" is the same concept as Backlinks and it shows links which point to the current note.

But you want this panel also show links which are "outgoing"? But based on a video Obisidian doesn't do that either - see e.g. https://youtu.be/n7xrHPpTWJ0?t=378 where the backlinks doesn't show the wellness link (since it's not a backlink).

**Comment by omega3** [comment]
Sorry for the confusion. I messed it up. 

Looks like it is a different concept putting words in brackets [[ ]] to create backlinks.
But both ideas are great and worth to implement although backlinks could replace current linking to notes, because backlinks seems to be more broader and convenient approach. 

I also was not precise when writing about switching back and forth, because it can be done with arrows that are in top left corner. But switching back and forth with arrows has limited usage.

So we can separate two concepts: 

1) two-way links to notes and considering that single note can contain links to many other notes "What links here pane" could contain all links mentioned in text. But then the tile of this pane would be a bit misleading. It would be not only "What links here" but also "where this note links to". Why to do that? When text is long and contains many links to other notes it could be useful to have all the links to other notes listed in one place, not necessarily in pane, but perhaps at the bottom of the note. Now, if I want to find what links are contained in this article I need to scroll through entire article.

2) Second, and maybe worth considering solution would be to resign from links to notes and replace it with [[backlinks]] as Obsidian and other wiki based applications do, because it more flexible and practical approach. 

To sum up, I hope you can consider [[backlinks]] approach. 

Backlinks are becoming more and more popular. Also Notion introduced backlinks. A lot of Roam Research alternatives have backlinks:
https://www.reddit.com/r/selfhosted/comments/f0z6yd/open_source_alternatives_to_roam_research/

**Comment by zadam** [comment]
> Two-way links to notes and considering that single note can contain links to many other notes "What links here pane" could contain all links mentioned in text. But then the tile of this pane would be a bit misleading. It would be not only "What links here" but also "where this note links to". Why to do that? When text is long and contains many links to other notes it could be useful to have all the links to other notes listed in one place, not necessarily in pane, but perhaps at the bottom of the note. Now, if I want to find what links are contained in this article I need to scroll through entire article.

For that I would recommend using the Link map which shows all the links, both incoming and outgoing.

But yeah, changing concept of "What links here" to more general "Links" would be possible. This widget was actually intended as a text only alternative to the Link map.

> Second, and maybe worth considering solution would be to resign from links to notes and replace it with \[\[backlinks\]\] to words in brackets \[\[ \]\]. They point to a specific position of word in a sentence and panel listing links in Obsidian list these backlink context sentences. It is more flexible and practical approach.

Here there are 2 separate things:

1.  syntax - I don't really see what using \[\[ and \]\] brings on top of normal links ...
2.  Obsidian displays a context around the link - here I agree that seeing the context in which the note is linked is useful and could be added.

**Comment by omega3** [comment]
First of all my apologies, because I now realized I misunderstood how Obsidian works. It also links to notes, to be exact to words that are the names of notes, in other words to notes headers.

> syntax - I don't really see what using [[ and ]] brings on top of normal links ...

[[ ]] looks like a common to this "Wiki" type of programs. But I admit Ctrl+L is equally good. So, let's forget about it. 

> "What links here" to more general "Links" would be possible. This widget was actually intended as a text only alternative to the Link map.

If you would do "What links here" and "where this note links to" as one pane "Links" there should be some indication that differentiates one from another for example:
→ "What links here"
← "where this note links to"

But this might be cluttered if there where many links or to put it in other words not giving very distinct visual indication.
Or alternatively, what about two panes. If I could hide "Link map" (for the purpose of having more space) and have two separate panes for "What links here" and "where this note links to", that would be a very convenient solution to me. 

Of course depending on space available in computer monitor, users could decide to have all panes visible. But with my small display being able to hide "Link map" would be a plus. 

My idea described here: "Second, and maybe worth considering solution would be to resign from links to notes and replace it with [[backlinks]] to words in brackets [[ ]]. They point to a specific position of word in a sentence and panel listing links in Obsidian list these backlink context sentences. It is more flexible and practical approach." would be completely another story. So, let's forget about it. I admit it wasn't well-thought-out.

**Comment by zadam** [comment]
All widgets in the right pane are "collapsible" - click on their title to toggle them. But I admit this isn't very discoverable - I just added an explicit button to toggle the visibility ...
